build_tools/github/trigger_hosting.sh

```diff
@@ -31,10 +31,29 @@ else
      BRANCH=$HEAD_BRANCH
 fi
 
-curl --request POST \
+# Circle CI REST API return HTTP response with 202 status code even when the POST requests fail.
+# Hence we add some handling so that errors are reported on GitHub.
+CIRCLE_CI_RESPONSE=$(curl --request POST \
      --url https://circleci.com/api/v2/project/gh/$GITHUB_REPOSITORY/pipeline \
      --header "Circle-Token: $CIRCLE_CI_TOKEN" \
      --header "content-type: application/json" \
      --header "x-attribution-actor-id: github_actions" \
      --header "x-attribution-login: github_actions" \
-     --data \{\"branch\":\"$BRANCH\",\"parameters\":\{\"GITHUB_RUN_URL\":\"$GITHUB_RUN_URL\"\}\}
+     --data \{\"branch\":\"$BRANCH\",\"parameters\":\{\"GITHUB_RUN_URL\":\"$GITHUB_RUN_URL\"\}\})
+
+echo CIRCLE_CI_RESPONSE
+
+CIRCLE_CI_RESPONSE_MESSAGE=$(cat $CIRCLE_CI_RESPONSE | jq ".message")
```

Comment:
According to CircleCI's [API docs](https://circleci.com/docs/api/v2/index.html#operation/triggerPipeline), if the message exist, then there was an error. In that case, we can use the existence of the message to know if the POST request was a success:

```bash
if [[ "$CIRCLE_CI_RESPONSE_MESSAGE" == "null" ]]; then
  exit 0  # No message means there was no error.
fi

if [[ "$CIRCLE_CI_RESPONSE_MESSAGE" =~ .*"Not Found".* ]]; then
  ...
fi

if [[ "$CIRCLE_CI_RESPONSE_MESSAGE" =~ .*"Permission denied".* ]]; then
  ...
fi

exit 1
```